### PR TITLE
Added finer grain capabilities

### PR DESCRIPTION
**Pull request by hbs**

### CODE REVIEW

**Review by pi-r-p — CHANGES_REQUESTED** [review]
These functions are added to WarpScript in a static block of FDBStoreClient.

Why isn't it in WarpScriptLib, to be consistent with leveldb functions ? 

Usage: 
In a standalone (leveldb), I may want to REXEC something to a fdb one, and this will fail:
```
<% 
  $x FDBGET   //will fail here, not able to build the function
%> { 'x' $tenant } DEREF  $url REXEC
```

**Comment by hbs** [comment]
The functions are only added when using FDB. If you want to do what you showed, use 'FDBGET' EVAL

**Comment by pi-r-p** [comment]
I know the workaround, but it is not as convenient. 

Functions are defined as a standard or by an extension, not by the backend type. And you already check the backend in all these functions, so it won't be a problem to treat them as leveldb ones.

In the current documentation process, it will more easy not to add this kind of exceptions.

**Comment by hbs** [comment]
Those functions were added in PR#1192 that you approved. We can modify the way they are defined but those types of concerns should be raised within the convo of the appropriate PR.

**Review by pi-r-p — CHANGES_REQUESTED** [review]
- `FDBSIZE` should have the same optional regexp whitelist as `FDBTENANT`.

Outside of this PR scope, but not spotted before :
- `FDBTENANT` returns an empty map when tenant does not exists. `FDBSIZE` raises an exception. It is not consistent.
- Is there a way to list all existing tenants ? (it could be very handy, and restricted to fdb.admin cap)

File: warp10/src/main/java/io/warp10/fdb/FDBSIZE.java [hunk 1]
```diff
@@ -34,8 +36,18 @@ public FDBSIZE(String name) {
...
+    if (null != Capabilities.get(stack, FDBUtils.CAPABILITY_ADMIN)) {
+      regexp = null;
+    } else if (null == Capabilities.get(stack, FDBUtils.CAPABILITY_SIZE)) {
+      regexp = Capabilities.get(stack, FDBUtils.CAPABILITY_SIZE);
+      // Empty string means match all tenants
+      if ("".equals(regexp)) {
+        regexp = null;
+      }
+    } else {
+      throw new WarpScriptException(getName() + " missing '" + FDBUtils.CAPABILITY_SIZE + "' or '" + FDBUtils.CAPABILITY_ADMIN + "' capability.");
```
Comment (pi-r-p) [inline warp10/src/main/java/io/warp10/fdb/FDBSIZE.java:50]: if you do not have CAPABILITY_ADMIN nor CAPABILITY_SIZE, you can read everything !

File: warp10/src/main/java/io/warp10/fdb/FDBTENANT.java [hunk 2]
```diff
@@ -32,8 +35,18 @@ public FDBTENANT(String name) {
...
+
+    if (null != Capabilities.get(stack, FDBUtils.CAPABILITY_ADMIN)) {
+      regexp = null;
+    } else if (null == Capabilities.get(stack, FDBUtils.CAPABILITY_TENANT)) {
+      regexp = Capabilities.get(stack, FDBUtils.CAPABILITY_TENANT);
+      // Empty string means match all tenants
+      if ("".equals(regexp)) {
+        regexp = null;
+      }
+    } else {
```
Comment (pi-r-p) [inline warp10/src/main/java/io/warp10/fdb/FDBTENANT.java:48]: Same wrong condition path, everything is allowed with no caps at all.

**Review by pi-r-p — CHANGES_REQUESTED** [review]

**Comment by hbs** [comment]
PTAL

**Review by pi-r-p — APPROVED** [review]

**Pull request merged**